tee-worker/litentry/core/identity-verification/src/web2/twitter/code_verifier_store.rs

```diff
@@ -0,0 +1,44 @@
+#[cfg(feature = "sgx")]
+use thiserror_sgx as thiserror;
+
+#[cfg(feature = "std")]
+use std::sync::RwLock;
+#[cfg(feature = "sgx")]
+use std::sync::SgxRwLock as RwLock;
+
+#[cfg(feature = "std")]
+use hex;
+#[cfg(feature = "sgx")]
+use hex_sgx as hex;
+
+use codec::Encode;
+use core::result::Result;
+use lazy_static::lazy_static;
+use std::{collections::HashMap, string::String, sync::Arc};
+
+use litentry_primitives::ParentchainAccountId as AccountId;
+
+lazy_static! {
+	static ref TWITTER_CODE_VERIFIER_STORE: Arc<RwLock<HashMap<String, String>>> =
+		Arc::new(RwLock::new(HashMap::new()));
+}
+
+pub struct CodeVerifierStore;
+
+impl CodeVerifierStore {
+	pub fn save_code(account_id: AccountId, code: String) -> Result<(), String> {
```

Comment:
Can we remove codes after some time? Otherwise this will grow and grow.

Reply:
The code is removed when consumed. I guess there could be some cases where the code was created but never consumed, for example if a user started the link identity process on the client but did not continue with the verification, but I'm not sure if this would be a problem. We would need to implement some sort of expire mechanism

Reply:
I suggest adding an expiration logic and a limit of N codes at time stored untill we find more memory safety storage.

Reply:
I changed the data structure to use LRU cache, when the data is consumed it is removed from the store and if the store hits the capacity limit the new values will replace the least recently used values.